products/views.py

```diff
@@ -1,3 +1,157 @@
-from django.shortcuts import render
+import random
+from django.core.exceptions import FieldError
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Q, query
+
+from .models import Category, Product, ProductFlavor
+
+def MakingList(queryset):
+    res_list = []
+    for product in queryset:
+
+        hash_tag        = list(product.tag.all().values("caption"))
+        first_image_url = product.images.all()[:1].get().image_url
+
+        res_list.append(
+            {
+                "id"               : product.id,
+                "name"             : product.name,
+                "price"            : product.price,
+                "dgree"            : product.dgree,
+                "ml"               : product.ml,
+                "awards"           : product.awards,
+                "tiny_description" : product.tiny_description,
+                "hash"             : hash_tag,
+                "grade"            : product.grade,
+                "image"            : first_image_url,
+                "expire_date"      : product.expire_date,
+                "keep"             : product.keep,
+                "category_name"    : product.category.name,
+            }
+        )
+    return res_list
+
+class ImageListView(View):
+    def get(self, request, product_id):
+        try:
+
+            images = list(Product.objects.get(id=product_id).images.all().values("image_url"))
```

Comment:
ImageView는 가독성과 데이터베이스 요청 쿼리 수를 줄이기 위해서 Image 객체부터 접근해서 가져올 수 있도록 수정해주세요~
```suggestion
            images = Image.objects...

```

Reply:
참고) 왜 작성하신 방법이 database 요청 수가 더 많은지 고민해보시면 좋을 것 같아요~